airbyte-integrations/connectors/destination-gcs-data-lake/src/main/kotlin/io/airbyte/integrations/destination/gcs_data_lake/dataflow/GcsDataLakeAggregate.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.gcs_data_lake.dataflow
+
+import io.airbyte.cdk.load.command.Dedupe
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.data.ArrayValue
+import io.airbyte.cdk.load.data.IntegerValue
+import io.airbyte.cdk.load.data.ObjectValue
+import io.airbyte.cdk.load.data.StringValue
+import io.airbyte.cdk.load.data.TimestampWithTimezoneValue
+import io.airbyte.cdk.load.data.TimestampWithoutTimezoneValue
+import io.airbyte.cdk.load.dataflow.aggregate.Aggregate
+import io.airbyte.cdk.load.dataflow.transform.RecordDTO
+import io.airbyte.cdk.load.toolkits.iceberg.parquet.io.Operation
+import io.airbyte.cdk.load.toolkits.iceberg.parquet.io.RecordWrapper
+import io.airbyte.cdk.load.util.serializeToString
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.math.BigInteger
+import java.time.ZoneOffset
+import org.apache.iceberg.Schema
+import org.apache.iceberg.Table
+import org.apache.iceberg.data.Record
+import org.apache.iceberg.io.BaseTaskWriter
+
+private val logger = KotlinLogging.logger {}
+
+class GcsDataLakeAggregate(
+    private val stream: DestinationStream,
+    private val table: Table,
+    private val schema: Schema,
+    private val stagingBranchName: String,
+    private val writer: BaseTaskWriter<Record>,
+) : Aggregate {
+    companion object {
+        val commitLock: Any = Any()
+    }
+
+    private val operationType =
+        if (stream.importType is Dedupe) {
+            Operation.UPDATE
+        } else {
+            Operation.INSERT
+        }
+
+    override fun accept(record: RecordDTO) {
+        // Convert RecordDTO to Iceberg Record
+        // Note: ValueCoercer has already nulled out-of-range integers in Parse stage
+        // Here we handle Iceberg-schema-specific conversions that depend on field types
+        val converter = io.airbyte.cdk.load.data.iceberg.parquet.AirbyteValueToIcebergRecord()
```

Comment:
Why are we re-creating the object `AirbyteValueToIcebergRecord` for each record thats coming in.